hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java

```diff
@@ -278,7 +278,8 @@ public abstract class Dialect implements ConversionContext, TypeContributor, Fun
 
 	private static final Pattern ESCAPE_CLOSING_COMMENT_PATTERN = Pattern.compile( "\\*/" );
 	private static final Pattern ESCAPE_OPENING_COMMENT_PATTERN = Pattern.compile( "/\\*" );
-	private static final Pattern QUERY_PATTERN = Pattern.compile( "^\\s*(select\\b.+?\\bfrom\\b.+?)(\\b(where|join)\\b.+?)$" );
+	private static final Pattern QUERY_PATTERN = Pattern.compile(
+		"^\\s*(select\\b.+?\\bfrom\\b.+?)(\\b(?:natural )?(?:left |right )?(?:inner |outer )?join.+?\\b)?(\\bwhere\\b.+?)$");
```

Comment:
According to e.g. [PostgreSQL docs](https://www.postgresql.org/docs/current/queries-table-expressions.html#id-1.5.6.6.5.6.4.3.1.1), this pattern does not cover all the possible join syntaxes.

Reply:
I have completed the regex with those 2 keywords : "cross" and "full"

Compliant queries :
```
select * from TEST join TEST2 using(X) where Y = 'foo';
select * from TEST left join TEST2 using(X) where Y = 'foo';
select * from TEST right join TEST2 using(X) where Y = 'foo';
select * from TEST full join TEST2 using(X) where Y = 'foo';
select * from TEST outer join TEST2 using(X) where Y = 'foo';
select * from TEST cross join TEST2 using(X) where Y = 'foo';
select * from TEST left outer join TEST2 using(X) where Y = 'foo';
select * from TEST right outer join TEST2 using(X) where Y = 'foo';
```

This regex aimed to detect the start of "join" query part, not to perfectly match its content.